src/main/java/com/bereda/external_api/service/ExternalExchangeRateApiService.java

```diff
@@ -1,32 +1,26 @@
 package com.bereda.external_api.service;
 
 import com.bereda.external_api.model.ExternalExchangeRateResponse;
-import org.springframework.beans.factory.annotation.Value;
+import lombok.AllArgsConstructor;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
 
 import java.util.Optional;
 
 @Service
+@AllArgsConstructor
 public class ExternalExchangeRateApiService {
 
     private final RestTemplate restTemplate;
     private final String baseUrl;
 
-    public ExternalExchangeRateApiService(final RestTemplate restTemplate, @Value("${apiRequest.url}") final String baseUrl) {
```

Comment:
same here - nive that you found how to create bean with baseUrl that it works but I wouldn't do it like that. What you had was good - don;t forget to remove not needed bean baseUrl after you revert changes